Extensions/TargetBlank/Filters/TargetBlankFilter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using OrchardCore.ResourceManagement;
+using System.Threading.Tasks;
+
+namespace Lombiq.HelpfulExtensions.Extensions.TargetBlank.Filters;
+public class TargetBlankFilter : IAsyncResultFilter
+{
+    private readonly IResourceManager _resourceManager;
+
+    public TargetBlankFilter(
+        IResourceManager resourceManager) => _resourceManager = resourceManager;
+
+    public async Task OnResultExecutionAsync(ResultExecutingContext context, ResultExecutionDelegate next)
+    {
+        if (context.Result is not ViewResult)
+        {
+            await next();
+
+            return;
+        }
+
+        const string script = @"<script>
+                                function targetBlank() {
+                                    const x=document.querySelectorAll('a');
+                                    for(let i=0;i<x.length;i++) {
+                                        if(!x[i].href.match(/^mailto:/)&&(x[i].hostname!==location.hostname&&(!x[i].href.match('javascript:')))) {
+                                            x[i].setAttribute('target','_blank')}}}
+                                window.addEventListener('load',function() {
+                                    window.setTimeout(targetBlank,100)},false);
+                               </script>";
```

Comment:
Let's make it more readable.

```suggestion
        const string script = @"
            <script>
                 function targetBlank() {
                     const x=document.querySelectorAll('a');
                     for(let i=0;i<x.length;i++) {
                         if(!x[i].href.match(/^mailto:/)&&(x[i].hostname!==location.hostname&&(!x[i].href.match('javascript:'))))
                         {
                             x[i].setAttribute('target','_blank');
                         }
                     }
                 }
                 window.addEventListener(
                     'load',
                     function() {
                         window.setTimeout(targetBlank,100)
                     },
                     false);
            </script>";
```

Reply:
You can simply press `Commit suggestion`, that way you won't make mistakes while copy and pasting for example here the lines between the script tags being idented by one more tab than in my suggestion. Please fix.

Reply:
Also, FYI, using [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind), this:

```js
function() {
    window.setTimeout(targetBlank,100)
}
```

can be replaced by the following statement, which achieves exactly the same:

```js
window.setTimeout.bind(window, targetBlank, 100)
```

Reply:
There is a comment already explaining why this isn't in a seperate `.js` file and when will it be in one, no need to waste time and resources on it now.